src/report/mod.rs

```diff
@@ -76,9 +94,12 @@ pub fn write_html_detail(
     diffs: &[String],
     rule_name: &str,
 ) -> Result<FileCompareResult, Error> {
+    let nominal_file_name = get_file_name(nominal.as_ref())?.to_string();
+    let actual_file_name = get_file_name(actual.as_ref())?.to_string();
+
     let mut result = FileCompareResult {
-        nominal: nominal.as_ref().to_string_lossy().to_string(),
-        actual: actual.as_ref().to_string_lossy().to_string(),
+        nominal: nominal_file_name,
+        actual: actual_file_name,
```

Comment:
if you assign to the struct field names, you can use the shortened field initializer syntax
```rust
let nominal = get_file(...);
let actual = get_file(...);

let result = FileCompareResult {
  nominal,
  actual,
  ...
};
```

Reply:
nominal/actual are name of function arguments. Instead, removed the variable and call the function directly in the struct:

`FileCompareResult {
        nominal: get_file_name(nominal.as_ref())?.to_string(),`